### ISSUE TITLE
Feature: serve_file

### ISSUE THREAD

**Issue by chris-b1** [JuliaWeb/HTTP.jl#567]
This seems like generic enough functionality it could make sense to put in HTTP.jl.  I don't have an api figured out, but just as a placeholder/starting point

```julia
# ?serve_file
#     serve_file(path; mimetype="infer", extension="infer", ...) :: HTTP.Response
#     serve_file(stream::HTTP.Stream, ...) :: Nothing

using HTTP

function handle_file(req::HTTP.Request)
    return serve_file("file.txt") 
end
```

Some pointers to other implementations

PkgServer - https://github.com/JuliaPackaging/PkgServer.jl/blob/065235d39b4f66018dbc51ce3cb6861429f095e2/src/resource.jl#L394

Genie - https://github.com/GenieFramework/Genie.jl/blob/899ea32d3a7a95aab3bfe508fc9d6821bde2a1e0/src/Router.jl#L992

Flask - https://github.com/pallets/flask/blob/632f85b65354ad573bb33c54a0a9dd3ffcef371f/src/flask/helpers.py#L447

**Comment by quinnj** [comment]
Yeah, I've thought this would be a useful "default" handler to have in place; I'd be in favor

**Comment by clarkevans** [comment]
See https://github.com/JuliaWeb/HTTP.jl/issues/653

**Comment by quinnj** [comment]
There's a solid implementation that go includes in their stdlib [here](https://go.dev/src/net/http/fs.go). If someone wanted to take a stab at translating to Julia, I think it would be fairly straightforward. I'm not going to mark this as 1.0 as it's an easy add-on post 1.0. If someone is interested, just comment here and I'll assign it; otherwise I'll get to this eventually in the future.

**Comment by quinnj** [comment]
HTTP now includes a fairly good static file/directory middleware by default
